### PR TITLE
hashsum/b2sum: when the checksum file is untagged, detect the size      

### PR DESCRIPTION
**Pull request by sylvestre**
Should fix tests/cksum/b2sum.sh

### CODE REVIEW

**Comment by cakebaker** [comment]
The test fails on Windows as there is no `/dev/null`.

File: src/uucore/src/lib/features/checksum.rs [hunk 1]
```diff
@@ -513,7 +512,16 @@ where
                     .unwrap_or((String::new(), None))
                 } else if let Some(a) = algo_name_input {
                     // When a specific algorithm name is input, use it and use the provided bits
-                    (a.to_lowercase(), length_input)
+                    // except when dealing with blake2b, where we will detect the length
+                    if algo_name_input == Some(ALGORITHM_OPTIONS_BLAKE2B) {
+                        let length = match expected_checksum.len() {
+                            128 | 96 | 64 | 48 | 32 => Some(expected_checksum.len() / 2),
```
Comment (cakebaker) [inline src/uucore/src/lib/features/checksum.rs:518]: I don't understand what you are doing on this line. Why do you divide by 2? And why are you only interested in some selected lengths?

Comment (sylvestre) [reply src/uucore/src/lib/features/checksum.rs]: selected length was a mistake :)
i added a comment about /2

File: tests/by-util/test_hashsum.rs [hunk 2]
```diff
@@ -890,3 +890,45 @@ fn test_star_to_start() {
...
+        "0e5751c026e543b2e8ab2eb06099daa1d1e5df47778f7787faab45cdf12fe3a8  f\n",
+        "4ded8c5fc8b12f3273f877ca585a44ad6503249a2b345d6d9c0e67d85bcb700db4178c0303e93b8f4ad758b8e2c9fd8b3d0c28e585f1928334bb77d36782e8  f\n",
+        "786a02f742015903c6c6fd852552d272912f4740e15847618a86e217f71f5419d25e1031afee585313896444934eb04b903a685b1448b755d56f701afe9be2ce  f\n",
+    ];
+
+    let mut checksum_content = String::new();
+    for &checksum in &checksums {
+        checksum_content.push_str(checksum);
+    }
+    at.write("ck", &checksum_content);
```
Comment (cakebaker) [inline tests/by-util/test_hashsum.rs:914]: You can simplify it to:
```suggestion
    at.write("ck", &checksums.join(""));
```

File: tests/by-util/test_hashsum.rs [hunk 3]
```diff
@@ -890,3 +890,41 @@ fn test_star_to_start() {
...
+        "cae66941d9efbd404e4d88758ea67670  f\n",
+        "246c0442cd564aced8145b8b60f1370aa7  f\n",
+        "0e5751c026e543b2e8ab2eb06099daa1d1e5df47778f7787faab45cdf12fe3a8  f\n",
+        "4ded8c5fc8b12f3273f877ca585a44ad6503249a2b345d6d9c0e67d85bcb700db4178c0303e93b8f4ad758b8e2c9fd8b3d0c28e585f1928334bb77d36782e8  f\n",
+        "786a02f742015903c6c6fd852552d272912f4740e15847618a86e217f71f5419d25e1031afee585313896444934eb04b903a685b1448b755d56f701afe9be2ce  f\n",
+    ];
+
+    at.write("ck", &checksums.join(""));
+
+    for _ in &checksums {
```
Comment (cakebaker) [inline tests/by-util/test_hashsum.rs:912]: This `for` loop seems to be unnecessary.

Comment (sylvestre) [reply tests/by-util/test_hashsum.rs]: oh yeah, artifact from my previous trial

**Review by cakebaker — APPROVED** [review]

**Comment by cakebaker** [comment]
> Congrats! The gnu test tests/cksum/b2sum is no longer failing!

Good work :)

**Pull request merged**